lang/antlr/PartiQLTokens.g4

```diff
@@ -362,13 +362,16 @@ IDENTIFIER_QUOTED
  */
 
 WS
-    : WHITESPACE+ -> skip;
+    : WHITESPACE+ -> channel(HIDDEN);
 
 COMMENT_SINGLE
-    : '--' ~[\r\n]* '\r'? '\n'? -> skip;
+    : '--' ~[\r\n]* '\r'? '\n'? -> channel(HIDDEN);
 
 COMMENT_BLOCK
-    : '/*' .*? '*/' -> skip;
+    : '/*' .*? '*/' -> channel(HIDDEN);
+
+UNRECOGNIZED
+    : . ;
```

Comment:
Could you please explain a little more on what does this means? 

The parser now could not parse Ion Literal, i.e., 1d0. After some experiment, removing this. fixed the issue. Not exactly sure why since I am not that familiar with ANTLR, but hopefully this can help you debug.

Reply:
See here
```
 Parser Error: unexpected token found, IDENTIFIER : '`1d0`'
```

Reply:
So, the Gradle ANTLR plugin loves to use the Gradle cache.

My output:
```
PartiQL> `1d0`
   | 
===' 
1
--- 
OK!
```

To remove your Gradle cache and run the CLI, run:
```
./gradlew clean                                 && \
rm -rf ~/.gradle/caches/build-cache-1/          && \
./gradlew clean assemble                        && \
./cli/shell.sh
```

Reply:
BTW, whenever the PartiQLTokens file has been updated and you see an error like the one you've seen, it's likely due to the Gradle cache.

Reply:
Also, for your first question -- since the token `UNRECOGNIZED` is at the bottom of the grammar (excluding ION mode), it's a catch-all for syntactic errors. If, in the middle of lexing, the lexer throws a Recognition Exception, we can't recover a single token. So, in the middle of typing something like:
```
1 + 'hello
```

without the second quotation, then we don't have access to any of the tokens, because it can't tokenize `'hello`.

So, the simplest way to get around this is an `UNRECOGNIZED` token that isn't used in the Parser. It just allows us to tokenize (then we only need to parse, which will throw a location exposing error). Another interesting thing about ANTLR is that the implementation will actually catch RecognitionExceptions (for lexing) and then just not consume the faulty token, and it'll just swallow the exception. So, it intentionally skips them. You can work around this by throwing an exception that isn't RecognitionException -- but then we're left with the inability to get tokens that can lex.